### PR TITLE
[RLlib] Fix MultiAgentEpisode getter bugs.

### PR DESCRIPTION
**Pull request by sven1977**
Fix MultiAgentEpisode getter bugs.
* Improved MultiAgentEpisode test cases.

e.g.:

```
        observations = [
            {"a0": 0, "a1": 0},
            {"a0": 1, "a1": 1},
            {"a1": 2},
            {"a1": 3},
            {"a1": 4},
        ]
        actions = observations[:-1]
        rewards = observations[:-1]
        episode = MultiAgentEpisode(
            observations=observations, actions=actions, rewards=rewards, len_lookback_buffer="auto"
        )
        episode.get_actions(0)  # <- should result in an index error (b/c all actions are in the lookback buffer by default), but didn't 
```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/env/utils/infinite_lookback_buffer.py [hunk 1]
```diff
@@ -464,7 +464,7 @@ def _get_int_index(
         # If index >= 0 -> Ignore lookback buffer.
         # Otherwise, include lookback buffer.
         if idx >= 0 or neg_indices_left_of_zero:
-            idx = self.lookback + idx - (_ignore_last_ts is True)
+            idx = self.lookback + idx
```
Comment (sven1977) [inline rllib/env/utils/infinite_lookback_buffer.py:467]: This is the actual bug fix.

Comment (simonsays1980) [inline rllib/env/utils/infinite_lookback_buffer.py:467]: Amazing how such a small modification changes the landscape completely.

File: rllib/env/tests/test_multi_agent_episode.py [hunk 2]
```diff
@@ -1023,6 +1023,50 @@ def test_get_actions(self):
         act = episode.get_actions(-4, env_steps=False, fill="skip")
         check(act, {"a0": "skip", "a1": 0})
 
+        episode.add_env_step(
```
Comment (sven1977) [inline rllib/env/tests/test_multi_agent_episode.py:1026]: Added this to the tests. Mostly to figure out, whether a hanging action at the edge of the episode or further back would make a difference in `get_actions(-1)`.

File: rllib/env/tests/test_multi_agent_episode.py [hunk 3]
```diff
@@ -975,7 +975,7 @@ def test_get_actions(self):
             check(act, actions[i])
         # Access >=0 integer indices (expect index error as everything is in
         # lookback buffer).
-        for i in range(1, 5):
+        for i in range(0, 5):
```
Comment (sven1977) [inline rllib/env/tests/test_multi_agent_episode.py:978]: idx=0 was NOT working properly before this fix.

File: rllib/env/multi_agent_episode.py [hunk 4]
```diff
@@ -1633,21 +1633,30 @@ def __repr__(self):
         )
 
     def print(self) -> None:
+        """Prints this MultiAgentEpisode as a table of observations for the agents."""
```
Comment (sven1977) [inline rllib/env/multi_agent_episode.py:1636]: Made this a little nicer. :)

**Review by simonsays1980 — APPROVED** [review]
LGTM. All tests passed for the MAERB, too. This was the remaining part in the puzzle. Awesome!

**Pull request merged**